testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OfflineTokenTest.java

```diff
@@ -743,6 +743,70 @@ public void offlineTokenBrowserFlowIdleTimeExpired() throws Exception {
         testOfflineSessionExpiration(IDLE_LIFESPAN, MAX_LIFESPAN, IDLE_LIFESPAN + SessionTimeoutHelper.IDLE_TIMEOUT_WINDOW_SECONDS + 60);
     }
 
+    // Issue 13706
+    @Test
+    public void offlineTokenReauthenticationWhenOfflinClientSessionExpired() throws Exception {
+        // expect that offline session expired by idle timeout
+        final int MAX_LIFESPAN = 360000;
+        final int IDLE_LIFESPAN = 900;
+        RealmRepresentation realmRep = adminClient.realm("test").toRepresentation();
+        int origIdleTimeout = realmRep.getSsoSessionIdleTimeout();
+        realmRep.setSsoSessionIdleTimeout(900);
+        adminClient.realm("test").update(realmRep);
+
+        getTestingClient().testing().setTestingInfinispanTimeService();
```

Comment:
I can add that `getTestingClient().testing().setTestingInfinispanTimeService()` is needed to reproduce the buggy scenario from the desciption of this bug, as without that, the infinispan "timer" is not integrated with Keycloak TimerProvider (setTimeOffset etc). And for this one particular scenario, it is needed to simulate the scenario when the item is not found in the infinispan cache after timeout.

The altertnative to explicit call `getTestingClient().testing().setTestingInfinispanTimeService()` is using the rule `InfinispanTestTimeServiceRule`, which can be declared on the class, but this causes some other tests of `OfflineTokenTest` class to fail. But if you prefer, I can add that rule and fix the other tests according to it. This might be more proper way (as in production, the infinispan timer and Keycloak TimerProvider would be also in sync), but will require more changes on the test class. WDYT?

Reply:
Let's keep it in the test as is